install/kubernetes/helm/istio/values.yaml

```diff
@@ -136,96 +136,105 @@ ingress:
       name: https
     selector:
       istio: ingress
+
 #
-# ingressgateway configuration
+# Ingress Gateway Configuration
+# By default (if enabled) a single Ingress Gateway will be created for the mesh.
+# Add additional configuration(s) to the `gateways` array to add more gateway(s). Make
+# sure those are uniquely named and that NodePorts are not duplicated.
 #
 ingressgateway:
   enabled: true
-  serviceAccountName: istio-ingressgateway-service-account
-  replicaCount: 1
-  autoscaleMin: 1
-  autoscaleMax: 1
-  resources: {}
-# limits:
-#  cpu: 100m
-#  memory: 128Mi
-# requests:
-#  cpu: 100m
-#  memory: 128Mi
-  service:
-    name: istio-ingressgateway #DNS addressible
-    labels:
-      istio: ingressgateway
-    annotations: {}
-    #namespace: istio-system
-    loadBalancerIP: ""
-    type: LoadBalancer #change to NodePort, ClusterIP or LoadBalancer if need be
-    ports:
-      ## You can add custom gateway ports
-    - port: 80
-      name: http
-      nodePort: 31380
-    - port: 443
-      name: https
-      nodePort: 31390
-    - port: 31400
-      name: tcp
-      nodePort: 31400
-  deployment:
-    labels:
-      istio: ingressgateway #will be added to pods and service
-    ports:
-    - containerPort: 80
-    - containerPort: 443
-    - containerPort: 31400
-    secretVolumes:
-    - name: ingressgateway-certs
-      secretName: istio-ingressgateway-certs
-      mountPath: /etc/istio/ingressgateway-certs
-    - name: ingressgateway-ca-certs
-      secretName: istio-ingressgateway-ca-certs
-      mountPath: /etc/istio/ingressgateway-ca-certs
+  gateways:
+  - serviceAccountName: istio-ingressgateway-service-account
+    replicaCount: 1
```

Comment:
Simplify this to the following:

```
gateways:
- name: istio-ingressgateway
   labels:
      istio: ingressgateway
   replica..
   type: Loadbalancer
   loadBalancerIP: ..
   annotations: {}
   ports:
   - name: http
      number: 80
      nodePort: ...
   secretVolumes:
   ...
    ....
 - name: istio-egressgateway
    ...
    type: ClusterIP
    labels:..
```

Essentially a single array of gateways. Every gateway here will be deployed. You can derive service name from the gateway.name, deployment labels from labels, service account from name, service ports from ports, container ports from service ports, etc. those who want more customization are typically knowledgeable enough to directly modify the service/deployment etc.

Reply:
Thought about it but wasn't bold enough to do such major changes. But now that you "approve" it..

Reply:
That would actually make things simple..

Reply:
Due to Helm parser limitations I can't have the array directly under `gateways`.
The workaround in my last commit has a `list` beneath it:

```yaml
gateways:
  enabled: true
  list:
  - name: istio-ingressgateway
    enabled: true
    labels:
      istio: ingressgateway
     ...
  - name: istio-egressgateway
    enabled: true
    labels:
      istio: egressgateway
    replicaCount: 1
    ...
```

Alternatively, it can be something like this:
```yaml
gateways:
  enabled: true
  istio-ingressgateway:
    enabled: true
    labels:
      istio: ingressgateway
  ...
  istio-egressgateway:
    enabled: true
    labels:
      istio: egressgateway
  ...
```

Reply:
Due to helm limitations mentioned above I decided to go with the second structure and committed those changes.